pkg/api/serve.go

```diff
@@ -40,26 +41,22 @@ func Serve(
 	gatewayDomain string,
 ) http.Handler {
 	logger.Info("initialize OpenAPI server")
-	swaggerSpec, _ := loads.Analyzed(restapi.SwaggerJSON, "")
-
-	api := operations.NewLakefsAPI(swaggerSpec)
-	api.Logger = func(msg string, ctx ...interface{}) {
-		logging.Default().WithField("logger", "swagger").Debugf(msg, ctx)
+	swagger, err := GetSwagger()
+	if err != nil {
+		panic(err)
 	}
-	api.BasicAuthAuth = NewBasicAuthHandler(authService)
-	api.JwtTokenAuth = NewJwtTokenAuthHandler(authService)
-
-	api.UseSwaggerUI()
-
-	api.ServeError = errors.ServeError
-
-	api.JSONConsumer = runtime.JSONConsumer()
-	api.MultipartformConsumer = runtime.DiscardConsumer
-
-	api.BinProducer = runtime.ByteStreamProducer()
-	api.JSONProducer = runtime.JSONProducer()
+	r := chi.NewRouter()
+	apiRouter := r.With(
+		OapiRequestValidatorWithOptions(swagger, &openapi3filter.Options{
+			AuthenticationFunc: openapi3filter.NoopAuthenticationFunc,
```

Comment:
Why not ride the authentication hook provided here instead of adding a custom middleware?
At least document the reason..